tokenizers/src/utils/regex.rs

```diff
@@ -0,0 +1,497 @@
+// All code and comments below this first line are copied from here (with some edits): https://github.com/bminixhofer/nlprule/blob/main/nlprule/src/utils/regex.rs
```

Comment:
Yes, overall refrain from doing things this way.

Copy pasting code from another repo is likely to be the wrong solution.
They have a different context, different goals, you can't just reuse and hope it fits the current projects needs.

Taking an inspiration from other sources is GREAT, and you should definitely continuing looking at other repos for ideas on how to handle problems.

But you cannot blindly apply things, you need to understand the current project scopes, philosophy, concerns.

Rule of thumb:
- Less changes is better
- Imitate the style of previous code is better